src/opm/output/eclipse/Summary.cpp

```diff
@@ -1380,54 +1380,29 @@ void eval_udq(const Opm::Schedule& schedule, std::size_t sim_step, Opm::SummaryS
     const UDQConfig& udq = schedule.getUDQConfig(sim_step);
     const auto& func_table = udq.function_table();
     UDQContext context(func_table, st);
-    {
-        const std::vector<std::string> wells = st.wells();
-
-        for (const auto& assign : udq.assignments(UDQVarType::WELL_VAR)) {
-            auto ws = assign.eval(wells);
-            for (const auto& well : wells) {
-                const auto& udq_value = ws[well];
-                if (udq_value)
-                    st.update_well_var(well, ws.name(), udq_value.value());
-            }
-        }
+    for (const auto& assign : udq.assignments(UDQVarType::WELL_VAR)) {
+        auto ws = assign.eval(st.wells());
+        st.update_udq(ws);
+    }
 
-        for (const auto& def : udq.definitions(UDQVarType::WELL_VAR)) {
-            auto ws = def.eval(context);
-            for (const auto& well : wells) {
-                const auto& udq_value = ws[well];
-                if (udq_value)
-                    st.update_well_var(well, def.keyword(), udq_value.value());
-            }
-        }
+    for (const auto& def : udq.definitions(UDQVarType::WELL_VAR)) {
+        auto ws = def.eval(context);
+        st.update_udq(ws);
     }
 
-    {
-        const std::vector<std::string> groups = st.groups();
-
-        for (const auto& assign : udq.assignments(UDQVarType::GROUP_VAR)) {
-            auto ws = assign.eval(groups);
-            for (const auto& group : groups) {
-                const auto& udq_value = ws[group];
-                if (udq_value)
-                    st.update_group_var(group, ws.name(), udq_value.value());
-            }
-        }
+    for (const auto& assign : udq.assignments(UDQVarType::GROUP_VAR)) {
+        auto ws = assign.eval(st.groups());
```

Comment:
This calls [`SummaryState::groups()`](https://github.com/OPM/opm-common/blob/67f935a502019c1c8a9ab38a6f94968aa8aa6e98/src/opm/parser/eclipse/EclipseState/Schedule/SummaryState.cpp#L258-L260) in a loop which creates a new `vector<string>`&mdash;albeit with the same values&mdash;for each call.  Does that have any noticeable performance impact?

Reply:
Yes - you are right; I thought of that. Have not measure performance impact - probably not *large* - but still worth caching I guess.